### DISCUSSION TITLE
Is there a way to fetch static / local data at build time on each page (_app / _document)?

### DISCUSSION THREAD

**Discussion by ConsoleTVs** [vercel/next.js#11158] [Help]
I want to make use of the static optimization to generate a full static website (without SSR). I have some function that loads some data and I would like to have it available on every page (or maybe mutate some context?). Is there a way to do that in nextjs? `getStaticProps` does not seem to work on _app or _document.

**Comment by timneutkens** [comment]
Duplicate of https://github.com/zeit/next.js/discussions/10949
